src/modeler/AutoRest.Swagger/Validation/ResourceRequiredMustNotHaveName.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ResourceRequiredMustNotHaveName : TypedRule<Dictionary<string, Schema>>
+    {
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "R2002";
```

Comment:
please add new fields to this class per https://github.com/Azure/autorest/blob/master/src/modeler/AutoRest.Swagger/Validation/Core/Rule.cs#L14
(ServiceDefinitionDocumentType , ValidationRuleMergeState, ValidationChangesImpact )